app/src/main/java/com/example/mvvmcleanarchitecture/features/characters/presentation/CharacterAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.mvvmcleanarchitecture.features.characters.presentation
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.mvvmcleanarchitecture.core.base.BaseAdapter
+import com.example.mvvmcleanarchitecture.databinding.ItemCharacterBinding
+import com.example.mvvmcleanarchitecture.features.characters.presentation.model.CharacterDisplayable
+
+class CharacterAdapter : BaseAdapter<CharacterDisplayable>() {
+    override val items: MutableList<CharacterDisplayable> by lazy { mutableListOf<CharacterDisplayable>() }
+    private lateinit var binding: ItemCharacterBinding
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        binding = ItemCharacterBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return CharacterViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val character = items[position]
+        (holder as CharacterViewHolder).bind(character)
+    }
+
+    class CharacterViewHolder(private val binding: ItemCharacterBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(character: CharacterDisplayable) {
+            binding.run {
+                Glide.with(root).load(character.image).into(imageView)
+                characterName.text = character.name
+                val speciesWithStatusLabel = "${character.status} - ${character.species}"
+                characterSpecies.text = speciesWithStatusLabel
```

Comment:
Mam jeszcze jedno pytanie: Jak wejdziemy na stronkę z api: https://rickandmortyapi.com/ to ukazuje nam się lista postaci, a obok statusu danej postaci kropka zamalowana na czerwono lub zielono. 
I teraz jeżeli chciałbym cos takiego dodać u siebie to powinienem:
- dodać klasę enum w warstwie domeny ktora by zawierala te statusy -> dead, alive itd.
-> konwertowac to na taką samą klasę do warstw danych i prezentacji -> 
-> w adapterze przy tworzeniu widoku (view holdera) wywolac metode ktora ustawi odpowiedni atrybut na widgecie w zaleznosci od tego statusu ?

Reply:
Możesz zrobić tak: 
- obiekt w warstwie danych pozostaje bez zmian, jedynie podczas konwersji na typ domenowy należy odpowiednio ustawić enum reprezentujący status 
- obiekt w warstwie domeny będzie zawierał enuma reprezentującego status - dead, alive, unknown 
- obiekt displayable może zawierać metodę / property reprezentującą kolor statusu np. `statusColor`. Gdy będziesz chciał ustawić taki kolor po stronie widoku nie będziesz musiał pisać żadnej dodatkowej logiki w adapterze, tylko po prostu pobierzesz kolor, a logika wyboru tego koloru będzie zaszyta w modelu displayable. 

Może to wyglądać dla przykładu tak: 
- dead -> kolor czerwony 
- alive -> kolor zielony 
- unknown -> kolor szary

Reply:
zrobię sobie to na ekranie detali w takim razie w 10 module.